### ISSUE TITLE
Whole Maze in main branch doesn't work.

### ISSUE THREAD

**Issue by avionml18** [avionml18/MM23#20]
When running Whole Maze algorithm in the Return To Start branch, it works, but in the main branch, it doesn't.

**Comment by avionml18** [comment]
The Whole Maze algorithm works sometimes. After running a few times, I noticed it would successfully run, so what's making the error happen at other times is still unknown.

**Comment by avionml18** [comment]
This issue has been fixed. 

The ndx_2_pop variable sometimes had an index that was out of range of the list but that index was part of a 2-pair valued system in the tuple_list. If I had scanned the list to find the matching index, that also would have fixed the problem.

How I fixed this was by changing the method of list from pop to remove, since I always had the value in hand or could get it fairly quickly. Adjusted code to accommodate only for coordinates in the tuple list and works consistently. 

![image](https://github.com/avionml18/MM23/assets/76172673/b0f75843-93b6-4e4f-84be-e44ed1e0183d)
![image](https://github.com/avionml18/MM23/assets/76172673/90a1b130-dcfd-412b-bc11-37eb589f743e)
